### ISSUE TITLE
id property added to all generated objects

### ISSUE THREAD

**Issue by rcoy-v** [json-schema-faker/json-schema-faker#113]
This property is added to all generated objects using faker:

`"id": "http://json-schema.org/schema#"`

This also conflicts with any objects that have an actual `id` property, causing an error. Is there anyway to disable this? The only other mention of this functionality I can find is #7.

**Comment by pateketrueke** [comment]
Please share your json-schema to better understanding

**Comment by rcoy-v** [comment]
This is a simple example using the online demo:

http://json-schema-faker.js.org/#%7B%22testProperty%22%3A%7B%22type%22%3A%22string%22%7D%7D

I would only expect `testProperty` to exist in the result object, but `id` exists as well.

**Comment by pateketrueke** [comment]
Well, in fact this is not a valid json-schema:

```json
{
  "testProperty": {
    "type": "string"
  }
}
```

Did you validate your schema before?

What result is you expecting from?

**Comment by rcoy-v** [comment]
We are extracting schemas defined in a swagger spec, which is validated with a parser. So expected what was passed to faker was also good. It seems like the end object  we are providing to faker isn't exactly in a valid format. So it still returns an expected fake object, but with that special `id` property attached. If I properly fix the provided schema, the `id` property is not appended. 

I think this is just a user issue, not passing valid input to faker.

**Comment by pateketrueke** [comment]
Interesting, I ran some tests and you're right: that schema is valid...

Actually the `id` is appended by `deref` in order to dereferencing sub-schemas according the spec.

I found this behavior strange and should be fixed, thanks!

**Comment by pateketrueke** [comment]
After some time researching I found this: http://www.jsonschemavalidator.net/

Is very useful to understand what "valid schema" means because this:

**invalid-schema.json**

```json
{
  "testProperty": {
    "type": "string"
  }
}
```

It's _valid_ but it cannot **validate** anything, e.g.

```json
{
  "testProperty": false
}
```

> The validator doesn't matter because the JSON-Schema is not well defined.

But this work as expected:

**valid-schema.json**

```json
{
  "type": "object",
  "properties": {
    "testProperty": {
      "type": "string"
    }
  }
}
```

> Immediately the validator warn about the incorrect data-type for `testProperty`:

> ```
> Found 1 error(s)
>
> Message:
>     Invalid type. Expected String but got Boolean.
> Schema path:
>     #/properties/testProperty/type
> ```

Of course `jsf` understood the given schema but the additional `id` is just a side-effect from the schema normalization.

I suggest you to normalize your schemas before passing them to `jsf`:

```js
var jsf = require('json-schema-faker');

var schema = {
  "testProperty": {
    "type": "string"
  }
};

var fixedSchema = {
  type: 'object',
  properties: schema,
  required: Object.keys(schema)
};

console.log(jsf(schema));
/*
{ testProperty: 'ipsum exercitation cillum',
  id: 'http://json-schema.org/schema#' }
*/

console.log(jsf(fixedSchema));
// { testProperty: 'cupidatat tempor in ex' }
```

What do you think?

**Comment by rcoy-v** [comment]
That very well summarizes what was happening on our end.

We were unknowingly passing only part of a defined schema; what was getting to faker was something very similar to the **invalid-schema.json**. That is when the `id` property was getting added. We didn't notice this at first since faker was still returning to us what looked like a good object at first.

When I found this, I fixed it to pass the entire schema. That looked much more like **valid-schema**.json. When I made this change, the `id` property was not attached.
